sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -3192,6 +3202,122 @@ atomic_update(T *p, Toffset offset, simd<T, N> src0, simd<T, N> src1,
       p, offset, src1, src0, mask);
 }
 
+template <native::lsc::atomic_op Op, typename T, int N, typename Toffset,
+          typename AccessorTy>
+__ESIMD_API std::enable_if_t<std::is_integral_v<Toffset> &&
+                                 __ESIMD_DNS::get_num_args<Op>() == 0 &&
+                                 !std::is_pointer<AccessorTy>::value,
+                             simd<T, N>>
+atomic_update(AccessorTy acc, simd<Toffset, N> offset, simd_mask<N> mask) {
```

Comment:
Why do we have this inconsistency between variants accepting offset as a) simd<Toffset,N>, b) simd_view<>, c) Toffset ?
I.e. why only variants (b) and (c) have the default value for the 'mask' operand?

It looks messy because you can call atomic_update(acc, simd<int,N>{0,4}.select<num,stride>()) without specifying a mask, but removing 'select' and passing offset as a simple simd object would start requiring the mask operand..

Reply:
why (c) would cause any problems? It is how to existing lsc_atomic_update() has proven to work, right?